chalk-integration/src/lowering.rs

```diff
@@ -529,6 +529,7 @@ impl LowerProgram for Program {
                                                     .intern(interner),
                                                 )
                                             })
+                                            .chain(opaque_ty.where_clauses.lower(&env)?)
```

Comment:
I don't think this is quite right. The *role* of the where-clauses is distinct from the *role* of the bounds. The bounds tell you what the hidden-type must satisfy; the where-clauses tell you what the generic parameters that the opaque type is used with must satisfy. They do not, in and of themselves, let you make assumptions. In particular, imagine an example like this one:

```rust
opauqe type Foo<T: Eq>: Debug = ...;
```

Here the `T: Eq` is a *where-clause* -- i.e., the idea is that `Foo<T>` is some type which is `Debug`, and it only works where `T: Eq`.

Reply:
Hmm, so these should then be a `Vec<TraitBound>` instead of a `<Vec<VariableKind>`, right? I'm not sure how to handle lowering for that correctly, since I don't think the current `LowerTypeKind` approach works for `TraitBound`'s. Would lowering for opaque types then be similar to `AssociatedTyLookup`?